### PR TITLE
Change logResult to only log once

### PR DESCRIPTION
**Pull request by mtyazici**
This fixes the issue https://github.com/hazelcast/hazelcast-aws/issues/222

However, there is a problem to consider. The cause of this log output is empty address list. This is not mutually exclusive with the errors `NoCredentialsException` and `RestClientException`. If address list is empty and one of the exceptions is thrown, user only will be notified by the empty address list. Upon fixing the empty address issue, they will see the other exceptions. We could solve this issue by using either an integer or an array instead of a boolean but I am not sure if the case I mentioned is a real life problem.

### CODE REVIEW

**Comment by leszko** [comment]
> This fixes the issue #222
> 
> However, there is a problem to consider. The cause of this log output is empty address list. This is not mutually exclusive with the errors `NoCredentialsException` and `RestClientException`. If address list is empty and one of the exceptions is thrown, user only will be notified by the empty address list. Upon fixing the empty address issue, they will see the other exceptions. We could solve this issue by using either an integer or an array instead of a boolean but I am not sure if the case I mentioned is a real life problem.

@mtyazici I think that the "emptyList" needs a separate bool variable. Because we may need to see both the exception logs and the empty list log.

**Comment by mtyazici** [comment]
> > This fixes the issue #222
> > However, there is a problem to consider. The cause of this log output is empty address list. This is not mutually exclusive with the errors `NoCredentialsException` and `RestClientException`. If address list is empty and one of the exceptions is thrown, user only will be notified by the empty address list. Upon fixing the empty address issue, they will see the other exceptions. We could solve this issue by using either an integer or an array instead of a boolean but I am not sure if the case I mentioned is a real life problem.
> 
> @mtyazici I think that the "emptyList" needs a separate bool variable. Because we may need to see both the exception logs and the empty list log.

Sure, I will change it .

**Review by leszko — APPROVED** [review]
LGTM 👍 

Please wait for the review from @alparslanavci and then feel free to merge.

**Review by alparslanavci — APPROVED** [review]

**Pull request merged**
